packages/terra-table/docs/guides/SectionWithinSection.md

```diff
@@ -0,0 +1,144 @@
+
+## State Management
+As section and subsection have the same API, we'll be walking through the expectation of a collapsible section in only one pattern.
+
+First defaulting our state to an empty array in the constructor. 
+```jsx
+  constructor(props) {
+    super(props);
+
+    this.state = { collapsedKeys: [] };
+  }
+```
+Next creating an event handler callback method to pass to the section's "onSelect" prop. The "onSelect" callback will return the metaData prop passed to each section.
+```jsx
+  handleSectionSelection(event, metaData) {
+
+  }
+```
+As a precaution we can prevent default on the event, in case the table has an ancestor with a listener. This also prevents the browser from auto page scrolling when we are intending to make a selection with the space bar.
+```jsx
+  handleSectionSelection(event, metaData) {
+    event.preventDefault();
+  }
+```
+Terra Table comes with additional helpers to manage state, in this case we want to determine if the selection has collapsed or opened the section using the section key in our state. So we use the utility method "updatedMulitSelectedKeys", which returns an array of the keys following the addition or removing of the key passed. We then set this as our state.
+```jsx
+  handleSectionSelection(event, metaData) {
+    event.preventDefault();
+    this.setState(state => ({ collapsedKeys: Utils.updatedMultiSelectedKeys(state.selectedKeys, metaData.key) }));
+  }
+```
+Settting state will trigger another render. So in the render method we need to generate our sections with the updated isCollapsed and isCollapsible props. Each section needs a unique key, not necessarily associated to our own key, but it works as well. The table renders flat, so keys need to be unique even if items are placed within sections structurally.
+[React List & Key Documentation](https://reactjs.org/docs/lists-and-keys.html)
+```jsx
+  createSection(sectionData) {
+    return (
+      <Section
+        key={sectionData.key}
+        title={sectionData.title}
+      >
+        {sectionData.childItems.map(childItem => createSubsection(childItem))}
+      </Section>
+    );
+  }
+```
+Next we need to set up our metaData object with our key value, and attach the "onSelect" to our handler.
+```jsx
+  createSection(sectionData) {
+    return (
+      <Section
+        key={sectionData.key}
+        title={sectionData.title}
+        metaData={{ key: sectionData.key }}
+        onSelect={this.handleSectionSelection}
+      >
+        {sectionData.childItems.map(childItem => createSubsection(childItem))}
+      </Section>
+    );
+  }
+```
+For the common collapsible sections we set "isCollapsible" for all sections.
```

Comment:
common collapsible sections  -> for rendering the initial collapsible section  maybe?

Reply:
Updated.